hal_espressif Kconfig can cause build to crashes if ZEPHYR_HAL_ESPRESSIF_MODULE_DIR is undefined

### Describe the bug

While writing a Dockerfile for a CI pipeline for zephyr I kept running into this error

```
/zephyr/scripts/kconfig/kconfig.py: Kconfig:8: recursive 'source' of 'Kconfig.zephyr' detected. Check that environment variables are set correctly.
Include path:
/bootloader/mcuboot/boot/zephyr/Kconfig:1486
Kconfig.zephyr:42
modules/Kconfig:137
modules/hal_espressif/Kconfig:10
Kconfig:8
```

Which was confusing since the same exact build worked on my own machine. What was even weirder was that we were not using an Espressif chip so it shouldn't have applied to this build. After doing some further research I found this line in ` zephyr/modules/hal_espressif/Kconfig`

`osource "$(ZEPHYR_HAL_ESPRESSIF_MODULE_DIR)/zephyr/Kconfig"`

since most normal zephyr setups aren't placed at root, if ZEPHYR_HAL_ESPRESSIF_MODULE_DIR is nothing, this skips and moves on. However, since my zephyr repo was at root in the docker image, it assumed it was pointing to /zephyr/Kconfig causing the recursion. 

My first though was to put it under the 

`if SOC_FAMILY_ESPRESSIF_ESP32` but upon running into the same error I found out that the if symbol does not stop sources from being used. Because of that I am unaware on how these changes should be fixed. For now I'll be setting up the folders in a more organized layout

this bug was introduced in this commit [248b8a7817b63306b1d35736df3054c9bb4db809](https://github.com/zephyrproject-rtos/zephyr/commit/248b8a7817b63306b1d35736df3054c9bb4db809)

### Regression

- [x] This is a regression.

### Steps to reproduce

Try and build a zephyr repo with zephyr at root

### Relevant log output

```shell
Parsing /bootloader/mcuboot/boot/zephyr/Kconfig
/zephyr/scripts/kconfig/kconfig.py: Kconfig:8: recursive 'source' of 'Kconfig.zephyr' detected. Check that environment variables are set correctly.
Include path:
/bootloader/mcuboot/boot/zephyr/Kconfig:1486
Kconfig.zephyr:42
modules/Kconfig:137
modules/hal_espressif/Kconfig:10
Kconfig:8
CMake Error at /zephyr/cmake/modules/kconfig.cmake:423 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /zephyr/cmake/modules/zephyr_default.cmake:129 (include)
  /zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:97 (include_boilerplate)
  CMakeLists.txt:18 (find_package)


-- Configuring incomplete, errors occurred!
CMake Error at cmake/modules/sysbuild_extensions.cmake:722 (message):
  CMake configure failed for Zephyr project: mcuboot

  Location: /bootloader/mcuboot/boot/zephyr
Call Stack (most recent call first):
  cmake/modules/sysbuild_images.cmake:43 (ExternalZephyrProject_Cmake)
  cmake/modules/sysbuild_default.cmake:21 (include)
  /zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:75 (include)
  /zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  /zephyr/share/sysbuild-package/cmake/SysbuildConfig.cmake:8 (include)
  template/CMakeLists.txt:10 (find_package)
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

The docker image was on Ubuntu 22.04, however I'm pretty sure it would work on any Unix system

### Additional Context

maybe we can format it like `zephyr/modules/hal_realtek/Kconfig`

--- a/modules/hal_espressif/Kconfig
+++ b/modules/hal_espressif/Kconfig
@@ -17,7 +17,7 @@ config ESP_HAL_EARLY_LOG_LEVEL
 	  These use esp_rom_printf and are independent of CONFIG_LOG.
 	  0 = Off, 1 = Error, 2 = Warning, 3 = Info, 4 = Debug
 
-endif # SOC_FAMILY_ESPRESSIF_ESP32
-
 # Include the HAL's own Kconfig
 osource "$(ZEPHYR_HAL_ESPRESSIF_MODULE_DIR)/zephyr/Kconfig"
+
+endif # SOC_FAMILY_ESPRESSIF_ESP32

